consensus/istanbul/core/core.go

```diff
@@ -387,33 +388,30 @@ func (c *core) waitForDesiredRound(r *big.Int) {
 
 	// Send round change
 	c.sendRoundChange(r)
+	return nil
 }
 
-func (c *core) resetRoundState(view *istanbul.View, validatorSet istanbul.ValidatorSet, nextProposer istanbul.Validator, roundChange bool) {
+func (c *core) resetRoundState(view *istanbul.View, validatorSet istanbul.ValidatorSet, nextProposer istanbul.Validator, roundChange bool) error {
 	// TODO(Joshua): Include desired round here.
-	if c.current == nil {
-		// When the current round is nil, we must start with the current validator set in the parent commits
-		// either `validatorSet` or `backend.Validators(lastProposal)` works here
-		c.current = newRoundState(view, validatorSet, nextProposer)
-	} else if roundChange {
-		c.current.StartNewRound(view.Round, validatorSet, nextProposer)
+	if roundChange {
+		return c.current.StartNewRound(view.Round, validatorSet, nextProposer)
+	}
+
+	// sequence change
+	var newParentCommits MessageSet
+	lastSubject, err := c.backend.LastSubject()
+	if err != nil && c.current.Proposal() != nil && c.current.Proposal().Hash() == lastSubject.Digest && c.current.Round().Cmp(lastSubject.View.Round) == 0 {
```

Comment:
if `err == nil`...

Reply:
I didn't change this line, only the indent changed:

The original is `err != nil`
```
if err != nil && c.current.Proposal() != nil && c.current.Proposal().Hash() == lastSubject.Digest && c.current.Round().Cmp(lastSubject.View.Round) == 0 {
```

Reply:
The original is wrong 😄